dashlivesim/dashlib/mpdprocessor.py

```diff
@@ -260,7 +262,7 @@ def process_patch(self, patch, mpd, mpd_data, period_data):
         segtimeline_generators = self.create_timeline_generators()
         last_period_id = '-1'
         for pdata in period_data:
-            if pdata.get('presentationTimeOffset') > self.patch_base:
+            if not segtimeline_generators and pdata.get('presentationTimeOffset') > self.patch_base:
```

Comment:
**I just figured out it is unnecessary, so I will remove this modification.**

Here is the reason that I added this.

Before I try patching_1, I tried without patching_1.
 http://vm2.dashif.org/livesim/periods_60/segtimeline_1/testpic_2s/Manifest.mpd

dash.js always crash at next line when moving to next period.
https://github.com/Dash-Industry-Forum/dash.js/blob/development/src/dash/models/DashManifestModel.js#L407

Because it will include Period with empty timelne like when moving to next period.
```xml
<Period id="p26320495" start="PT1579229700S">
      <AdaptationSet contentType="audio" id="1" lang="eng" mimeType="audio/mp4" segmentAlignment="true" startWithSAP="1">
         <Role schemeIdUri="urn:mpeg:dash:role:2011" value="main" />
         <SegmentTemplate initialization="$RepresentationID$/init.mp4" media="$RepresentationID$/t$Time$.m4s" presentationTimeOffset="75803025600000" timescale="48000">
<SegmentTimeline>
</SegmentTimeline>
</SegmentTemplate>
         <Representation audioSamplingRate="48000" bandwidth="48000" codecs="mp4a.40.2" id="A48">
            <AudioChannelConfiguration schemeIdUri="urn:mpeg:dash:23003:3:audio_channel_configuration:2011" value="2" />
         </Representation>
      </AdaptationSet>
      <AdaptationSet contentType="video" id="2" maxFrameRate="60/2" maxHeight="360" maxWidth="640" mimeType="video/mp4" minHeight="360" minWidth="640" par="16:9" segmentAlignment="true" startWithSAP="1">
         <Role schemeIdUri="urn:mpeg:dash:role:2011" value="main" />
         <SegmentTemplate initialization="$RepresentationID$/init.mp4" media="$RepresentationID$/t$Time$.m4s" presentationTimeOffset="142130673000000" timescale="90000">
<SegmentTimeline>
</SegmentTimeline>
</SegmentTemplate>
         <Representation bandwidth="300000" codecs="avc1.64001e" frameRate="60/2" height="360" id="V300" sar="1:1" width="640" />
      </AdaptationSet>
   </Period>
```

Crash happens [here](https://github.com/Dash-Industry-Forum/dash.js/blob/development/src/dash/models/DashManifestModel.js#L407).

First, I though it is a bug on dash.js, but DASH specification doesn't seems to allow empty SegmentTimeline.

```xml
<!-- Segment Timeline --> <xs:complexType name="SegmentTimelineType"> <xs:sequence> <xs:element name="S" minOccurs="1" maxOccurs="unbounded" > <xs:complexType> <xs:attribute name="t" type="xs:unsignedLong"/> <xs:attribute name="d" type="xs:unsignedLong" use="required"/> <xs:attribute name="r" type="xs:int" use="optional" default="0"/> <xs:anyAttribute namespace="##other" processContents="lax"/> </xs:complexType> </xs:element> <xs:any namespace="##other" processContents="lax" minOccurs="0" maxOccurs="unbounded"/> </xs:sequence> <xs:anyAttribute namespace="##other" processContents="lax"/> </xs:complexType>
```

That's why I was trying to fix simulator side.

I thought the newest period is causing it but I just found the oldest period became such a period with empty SegmentTimeline and it is causing this crash.

When it is using patch mode, the oldest period never changes on client in memory buffer as you know, so this crash won't happen when using patching_1.

Reply:
Removed it.
https://github.com/technogeek00/dash-live-source-simulator/pull/1/commits/4214caaed5731349286d985fc4001ae7376e3466